src/index.js

```diff
@@ -58,6 +70,8 @@ export default ({
         }
       }, ...transformResponse],
       ...rest
+    }).then(() => {
+      finished = true
```

Comment:
if you don't return the result here, it's will always be `undefined` or am I missing something?

Also you probably need to do
```js
.catch(err => {
  finished = true
  throw err
})
```
as well?